osm2gtfs/core/elements.py

```diff
@@ -0,0 +1,177 @@
+# coding=utf-8
+
+import sys
+import attr
+
+
+@attr.s
+class Element(object):
+    """The basic data element.
+    Contains the common attributes all other data classes share.
+
+    """
+
+    osm_id = attr.ib()
+    osm_type = attr.ib()
+    osm_url = attr.ib()
+
+    tags = attr.ib()
+    name = attr.ib()
+
+
+@attr.s
+class Line(Element):
+    """A general public transport service Line.
+
+    It's a container of meta information and different Itinerary objects for
+    variants of the same service line.
+
+    In OpenStreetMap this is usually represented as "route_master" relation.
+    In GTFS this is usually represented as "route".
+
+    """
+    route_id = attr.ib()
+
+    route_type = attr.ib(default=None)
+    route_desc = attr.ib(default=None)
+    route_color = attr.ib(default="#FFFFFF")
+    route_text_color = attr.ib(default="#000000")
+
+    # Related route variants
+    _itineraries = attr.ib(default=attr.Factory(list))
+
+    def __attrs_post_init__(self):
+        '''
+        Populates the object with information obtained from the tags
+        '''
+        # pylint: disable=unsupported-membership-test,unsubscriptable-object
+        if 'colour' in self.tags:
+            self.route_color = self.tags['colour']
+
+        # pylint: disable=unsupported-membership-test,unsubscriptable-object
+        if 'ref:colour_tx' in self.tags:
+            self.route_text_color = self.tags['ref:colour_tx']
+
+        # pylint: disable=unsupported-membership-test,unsubscriptable-object
+        if 'route_master' in self.tags:
+            self.route_type = self.tags['route_master'].capitalize()
+        elif 'route' in self.tags:
+            self.route_type = self.tags['route'].capitalize()
+        else:
+            self.route_type = "Bus"
```

Comment:
We should add a log for the route_master relations that do not have route_master tag: this is an issue that should be fixed in OSM.

Reply:
Yes, makes sense. Added.